configmap-watcher/cmd/cmd_test.go

```diff
@@ -0,0 +1,194 @@
+package cmd_test
+
+import (
+	"context"
+	"errors"
+	"os"
+	"testing"
+	"time"
+
+	"go.goms.io/aks/configmap-watcher/cmd"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/kubernetes"
+	testclient "k8s.io/client-go/kubernetes/fake"
+	"k8s.io/client-go/kubernetes/scheme"
+	k8stesting "k8s.io/client-go/testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestSuccessCommandNoConfigmap(t *testing.T) {
+	var cli cmd.KubeClient = &KubectlMock{}
+	_, err := cli.CreateClientSet("kubeconfig-file", "user-agent")
+	if err != nil {
+		t.Fatalf("Error creating fake client: %v", err)
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+	tmpDir := t.TempDir()
+
+	rootCmd := cmd.NewKubeCommand(cli)
+	rootCmd.SetArgs([]string{
+		"--kubeconfig-file=/config/fake/kubeconfig",
+		"--settings-volume=" + tmpDir,
+		"--configmap-name=ama-metrics-settings-configmap",
+		"--configmap-namespace=kube-system",
+	})
+
+	go func() {
+		if err := rootCmd.ExecuteContext(ctx); err != nil && !errors.Is(ctx.Err(), context.DeadlineExceeded) {
+			t.Error("Command execution failed: %w", err)
+			return
+		}
+	}()
+
+	// Wait for the context to be done (either by command completion or timeout)
+	<-ctx.Done()
+}
+
+func TestInvalidParameters(t *testing.T) {
```

Comment:
not sure if it makes sense here or in watch_test, but we should have tests for handling any of the customer configurable parts of the configmap being invalid / malformed

Reply:
This is needed here because is testing the parameters from the command itself. But you are right and we need that other test! reminds me I still haven't tested the configmap validator so will give it a try  and add an unit test in a separate PR